dist/bin/scala

```diff
@@ -31,7 +31,10 @@ source "$PROG_HOME/bin/common"
 # exec here would prevent onExit from being called, leaving terminal in unusable state
 compilerJavaClasspathArgs
 [ -z "${ConEmuPID-}" -o -n "${cygwin-}" ] && export MSYSTEM= PWD= # workaround for #12405
-eval "\"$JAVACMD\"" "-Dscala.home=$PROG_HOME" "-classpath \"$jvm_cp_args\"" "dotty.tools.MainGenericRunner" "-classpath \"$jvm_cp_args\"" "$@"
+echo "$@" | xargs -d' ' -n 1 | grep "^[^-J]"
+java_args=`echo "$@" | xargs -d' ' -n 1 | grep "^-J" | cut -c 3- | tr '\n' ' '`
```

Comment:
This looks rather tricky, can we parse the options like Scala 2 does instead? https://github.com/scala/scala/blob/2.13.x/src/compiler/templates/tool-unix.tmpl#L130-L147

Reply:
Will change, thanks